### PR TITLE
xpack-security-role-and-role-mapping-api

### PR DESCRIPTION
**Pull request by Enteris**
Added support for XPack Info API
Added support for XPack Security Role and Role Mapping API.

### CODE REVIEW

**Comment by Enteris** [comment]
There was an issue in the golang 1.11 run with module. 
https://travis-ci.org/olivere/elastic/jobs/447922832
Not sure it's related to my changes though. Could you confirm if I need to do anything about that?

**Comment by olivere** [comment]
> There was an issue in the golang 1.11 run with module.
> https://travis-ci.org/olivere/elastic/jobs/447922832
> Not sure it's related to my changes though. Could you confirm if I need to do anything about that?

No, that probably was a fault with getting the dependencies via `go get` at that time. That happens sometimes.

File: xpack_info.go [hunk 1]
```diff
@@ -0,0 +1,175 @@
+// Copyright 2005-2018 Coveo Solution. All rights reserved.
```
Comment (olivere) [inline xpack_info.go]: IANAL but this might probably be a problem. I don't want to run into any problems due to legal issues; it's just not worth it.

**Comment by Enteris** [comment]
I’ll take a look at it im not sure either what is implied by that copyright statement. I’ll come back to you with more info soon. 



Le 2 nov. 2018 à 02:36, Oliver Eilhard <notifications@github.com> a écrit :




@olivere requested changes on this pull request.
This code is looking good. Thanks you for working on it.
Re the copyright issue: I'm not sure how a copyright statement influences a license as IANAL. Personally, I don't care what you do with this library, as long as I'm not sued. I looked up other libraries and they sometimes have different copyright holders
 but a single licenses. As far as I understand it, copyright cannot be changed anyway, at least in most western countries. Again, IANAL, and I don't have the time and resources to talk to one.

In 
xpack_info.go:
> @@ -0,0 +1,175 @@
+// Copyright 2005-2018 Coveo Solution. All rights reserved.

IANAL but this might probably be a problem. I don't want to run into any problems due to legal issues; it's just not worth it.
—
You are receiving this because you authored the thread.
Reply to this email directly, 
view it on GitHub, or 
mute the thread.

File: xpack_info.go (hunk 1 above)
Comment (Enteris) [reply xpack_info.go]: I'm looking with our legal dep about what I can do. If they are alright with it I'll just change it back to whatever is okay with everyone.

**Comment by olivere** [comment]
Yeah, sorry for the inconvenience. I'm not propagating that it's my code, but I'm unsure of the consequences of using a company name in the Copyright.

I would love to add a header that states ...

```
// Copyright (c) 20xx The authors of "elastic" library.
```

... and then just refer to the list of contributors.

**Comment by Enteris** [comment]
I personally really don't mind what's written there. I just don't want trouble with my company. If they are fine with having either my name, your name, a contributor licence or just nothing at all written there, it will satisfy me. I'll keep you posted. I don't like legal stuff :crying_cat_face:

**Comment by Enteris** [comment]
@olivere Alright, got in touch with legal and they told me to keep it simple. I'll revert the licensing to what it was before (after that if you want to change it for the Contributor you may) and we'll call it a day :+1: I'll fix a few doc link and re-commit (hopefully the CI will pass this time)

**Comment by olivere** [comment]
Happy to hear! Sorry for all that ping-pong.

**Comment by Enteris** [comment]
No worry, I'm kinda new to Open Source contribution and it's always better to be safe than sorry! Cheers.

**Comment by olivere** [comment]
Oh sorry. Haven't seen you already updated it. Will look at it tomorrow, otherwise during Goto conference next week.

**Pull request merged**